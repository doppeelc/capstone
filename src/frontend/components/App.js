import { useEffect, useState } from 'react';
import Routes from './Routes';
import UserContext from './UserContext';
import NavBar from './NavBar';
import UserApi from '../api';
import { jwtDecode } from "jwt-decode";
import '../styles/App.css';
import FactsApi from '../factsApi';

function App() {

    const [ token, setToken ] = useState(localStorage.getItem("token"));
    const [ currentUser, setCurrentUser ] = useState({
        loaded:false,
        data:null
    });

    useEffect(
        function loadUser() {
    
            async function getUser() {
                if(token) {
                try {
                    let { username } = jwtDecode(token);
                    JoblyApi.token = token;
                    let user = await JoblyApi.getCurrentUser(username);
                    setCurrentUser({loaded:true, data:user});
                    setApplications(user.applications);
                } catch (e) {
                    setCurrentUser({loaded:true, data:null});
                }
                } else {
                    setCurrentUser({loaded:true, data:null});
                }
            }
            getUser();
        },
        [token]
    );

    async function signup(user) {
      let res = await UserApi.register(user);
      setToken(res);
      localStorage.setItem("token", res);
    }
  
    async function login(username, password) {
      let res = await UserApi.login(username, password);
      setToken(res);
      localStorage.setItem("token", res);
    }
  
    function logout() {
      setCurrentUser({loaded:true, data:null});
      setToken(null);
      localStorage.removeItem("token");
    }
  
    async function getNumberFact() {
      let fact = await FactsApi.getNumbersFact();
      return fact;
    }

    async function getAdvice() {
        let advice = await FactsApi.getAdvice();
        return advice;
    }

    async function getChuckJoke() {
        let joke = await FactsApi.getChuckJoke();
        return joke;
    }
  
    if(!currentUser.loaded) {
      return <p>Loading...</p>
    }

  return (
    <div className="App">
        <UserContext.Provider value={{
            currentUser:currentUser,
            number:{getNumberFact} ,
            advice:{getAdvice},
            chuck:{getChuckJoke}
        }}>
            <NavBar currentUser={currentUser.data}
                    logout={logout} />

            <Routes currentUser={currentUser.data}
                    signup={signup}
                    login={login} />
        </UserContext.Provider>
    </div>
  );
}

export default App;
