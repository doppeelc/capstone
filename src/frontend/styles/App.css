.App {
    text-align: center;
}

.App-error {
    text-align: center;
    height: 50px;
    margin-bottom: 40px;
}

.App-error-text {
    font-size: 20px;
    font-weight: bolder;
    background-color: lightcoral;
    padding: 20px;
    margin-top: 0px;
}