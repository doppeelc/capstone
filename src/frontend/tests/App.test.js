import { render } from '@testing-library/react';
import App from '../components/App';

test('renders without crashing', () => {
    render(<App />);
});

test('matches the snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
});






