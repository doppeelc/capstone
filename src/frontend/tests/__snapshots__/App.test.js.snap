// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md"
    id="NavBar"
  >
    <ul
      class="nav"
      id="NavBar-Nav"
    >
      <h1
        id="NavBar-center"
      >
        Factly
      </h1>
      <li
        class="nav-item"
        data-testid="withoutUser"
        id="NavBar-right"
      >
        <a
          href="/login"
          id="NavBar-link"
        >
          Login
        </a>
      </li>
      <li
        class="nav-item"
        id="NavBar-right"
      >
        <a
          href="/signup"
          id="NavBar-link"
        >
          Sign Up
        </a>
      </li>
    </ul>
  </nav>
  <div>
    <h3>
      Please Sign Up or Log In.
    </h3>
  </div>
</DocumentFragment>
`;
